### ISSUE TITLE
[Tribute] Cannot bind to DIV in test

### ISSUE THREAD

**Issue by JakeBeresford** [zurb/tribute#415]
I am using tribute ina vue JS project with the vue-tribute wrapper, I have a fairly simple MentionBox component that has a contenteditable DIV and some simple actions.

Everything is working nicely, until I started to write some Jest tests for the MentionBox component. I mostly want to test against those actions, but I am not able to mount the component as I get an error from this library

```
Error in mounted hook: "Error: [Tribute] Cannot bind to DIV"
```
This seems to come from here: https://github.com/zurb/tribute/blob/master/src/Tribute.js#L194

I am wondering whether this is a known issue when testing tribute, whether this might be a limitation of Jest, or if there is a known workaround.

Thanks for your time, this is a really great little library!

**Comment by mrsweaters** [comment]
Thanks for reporting! Do you have `contenteditable` on the element? Can you post a codepen?

**Comment by JakeBeresford** [comment]
I do have `contenteditable` on the element, my component is working everywhere except tests, running jest v24.9. 

No codepen (too much setup) but this is the current code for my component:

`
<template>
  <div class="mention-box" :class="{'mention-box--open': isOpen}">
    <p
      v-if="!isOpen"
      class="mention-box__placeholder"
      @click="openMentionbox"
    >
      {{ placeholder }}
    </p>
    <vue-tribute v-else ref="vTribute" :options="tributeOptions">
      <div
        ref="mentionBox"
        class="mention-box__input"
        role="textbox"
        contenteditable="true"
        aria-multiline="true"
        @input="updateMentionContent"
      />
    </vue-tribute>

    <div v-if="isOpen" class="mention-box__actions">
      <button
        class="mention-box__button mention-box__button-mention"
        @click="insertMention"
      >
        <icon name="at" class="mention-box__button-icon" />
        Mention someone
      </button>

      <button
        class="mention-box__button mention-box__button-mention"
        @click="uploadImage"
      >
        <icon name="photo" class="mention-box__button-icon" />
        Upload image
      </button>
    </div>
  </div>
</template>

<script>
import VueTribute from 'vue-tribute';
import Icon from '@/components/Global/Icon.vue';
import { imageFromContext } from '@/utils/image-helpers';
import gql from 'graphql-tag';

const USERS_FOR_MENTIONS_QUERY = gql`
query users($text: String) {
  userSearch(username: $text) {
    id
    username: name
    avatarSlugThumb
  }
}
`;

export default {
  name: 'MentionBox',
  components: {
    VueTribute,
    Icon,
  },
  props: {
    value: {
      type: String,
      default: '',
    },
    placeholder: {
      type: String,
      default: 'Write a comment...',
    },
  },
  data() {
    return {
      isOpen: false,
      mentionLimit: 3,
      mentionContent: '',
      tributeOptions: {
        menuContainer: this.$refs.mentionBox,
        containerClass: 'user-suggestions',
        selectClass: 'suggested-user--highlighted',
        menuItemTemplate(item) {
          return `
            <div class="suggested-user">
              <img class="suggested-user__image" src="${item.original.avatarSlugThumb || imageFromContext('default-user-medium.png')}" alt="Picture of ${item.original.username}" />
              <span class="suggested-user__name">${item.original.username}</span>
            </div>`;
        },
        selectTemplate(item) {
          return `<span class="mention-box__mention" data-user-id="${item.original.id}">${item.original.username}</span>`;
        },
        lookup: 'username',
        values: this.getUserSuggestions,
      },
    };
  },
  computed: {
    mentionCount() {
      const pattern = /<span class="mention-box__mention"/;
      const mentions = this.mentionContent.match(new RegExp(pattern, 'g'));
      return mentions ? mentions.length : 0;
    },
  },
  created() {
    if (this.value.length > 0) { this.isOpen = true; }
  },
  mounted() {
    if (this.isOpen) { this.$refs.mentionBox.innerText = this.value; }
  },
  methods: {
    openMentionbox() {
      this.isOpen = true;
      this.$nextTick(() => {
        this.$refs.mentionBox.focus();
      });
    },
    insertMention() {
      this.$refs.vTribute.tribute.showMenuForCollection(this.$refs.mentionBox);
    },
    uploadImage() {
      // TODO: Implement image uploads
      console.log('begin image upload...');
    },
    mentionsToShortcodes() {
      const pattern = /<span class="mention-box__mention" data-user-id="([a-zA-z0-9]*)">([a-zA-z0-9\s]*)<\/span>/;
      return this.mentionContent.replace(pattern, '[mention username="$2" id="$1"]');
    },
    updateMentionContent(event) {
      this.mentionContent = event.target.innerHTML;
      this.$emit('input', this.mentionsToShortcodes());
    },
    getUserSuggestions(text, cb) {
      if (this.mentionCount >= this.mentionLimit) {
        cb([]);
        return;
      }
      this.$apollo.query({
        query: USERS_FOR_MENTIONS_QUERY,
        variables: {
          text,
        },
      }).then((response) => {
        cb(response.data.userSearch);
      });
    },
  },
};
</script>
`

And my current test is about as simple as can be
`
import { mount } from '@vue/test-utils';
import MentionBox from '@/components/Mention/MentionBox.vue';

const component = (propsData) => mount(MentionBox, { propsData });

describe('MentionBox.vue', () => {
  it('Renders a MentionBox', () => {
    const wrapper = component();

    expect(wrapper.html()).toMatchSnapshot();
  });
}
`
If you have any suggestions I'm happy to try anything

**Comment by mrsweaters** [comment]
Can you try this branch in your code: https://github.com/zurb/tribute/pull/421

It might help. It looks like Jest uses JSDOM internally which has issues with `contentEditable`: https://github.com/jsdom/jsdom/issues/1670

**Comment by JakeBeresford** [comment]
Hey @mrsweaters - sorry for the slow response on this, I just came back to my component and was hoping to test it. I installed tribute from your branch and re-ran my spec but hit the same error with `[Vue warn]: Error in mounted hook: "Error: [Tribute] Cannot bind to DIV"`

I confirmed that the version of tribute in my node_modules includes your change in node_modules/tributejs/src/Tribute.js line 222

Not sure what else I can do to help here?

**Comment by mrsweaters** [comment]
Yeah I'm not sure how I can help either as I can't reproduce the issue. Thank you for the follow up information and I'll keep this issue open so others with a similar issue can report.

**Comment by lcwalther** [comment]
I'm using tribute in a React project and I'm encountering the same problem while writing tests in jest. I created a codesandbox in which the error is happening: https://codesandbox.io/s/xenodochial-mirzakhani-vu0y1?file=/src/App.test.js
If you need any further information, please let me know.

**Comment by mrsweaters** [comment]
This is helpful, thank you!

**Comment by rupert-ong** [comment]
Any word on merging this branch into master? Thanks.

**Comment by oofdie** [comment]
I also encountered the same issue in a React project where I used Tribute. I'm using jest + enzyme, and when I used mount for deep rendering, I encountered the problem `[Tribute] Cannot bind to DIV`.

**Comment by oofdie** [comment]
I solved this problem in a somewhat hacky way. I passed rootRef.current into the attach function to provide the DOM to the Tribute instance. Upon examining the source code of Tribute, I found that this error only occurs when dom.contentEditable is not true (even though I had already defined this property as true in the render function). So, I manually set this property to true like this: rootRef.current.contentEditable = true. After doing this, the problem was resolved. I hope this can help developers who are facing the same issue and temporarily get your test cases running.
